code/game/objects/structures/bonfire.dm

```diff
@@ -241,7 +241,15 @@
 		if(!consume_fuel(DEFAULTPICK(contents, null)))
 			extinguish()
 			return
-	if(!grill)
+	if(grill)
+		for(var/obj/item/I in get_turf(src))
+			if(istype(I, /obj/item/reagent_containers/glass/food_holder))
+				var/obj/item/reagent_containers/glass/food_holder/FH = I
+				FH.tick_heat(2 SECONDS, HEAT_MID, METHOD_GRILL)
```

Comment:
would using delta_time make sense here?

Reply:
most players will not be timing how long they're cooking for externally, and will be relying on checking it every so often in-game.

as such, making it a fixed amount per tick makes cooking more usable during high-lag scenarios, at the cost of being slowed down in these instances. IMO it's worth the trade-off